File: Python/algorithmic_toolbox/06_least_common_multiple.py
import sys


def convert_to_intlist(line):
    return [int(elem) for elem in line.split()]


def calc_gcd(a, b):
    while b > 0:
        a, b = b, a % b
    return a


def calc_lcm(a, b):
    gcd = calc_gcd(a, b)
    return (a // gcd) * b


def main():
    data = sys.stdin.read().splitlines()
    a, b = convert_to_intlist(data[0])
    result = calc_lcm(a, b)
    print(result)


if __name__ == "__main__":
    main()

File: Julia/algorithmic_toolbox/06-least-common-multiple.jl
converttointlist(line) = map(x -> parse(Int, x), split(line))


function calcgcd(a, b)
    while b > 0
        a, b = b, a % b
    end
    a
end


function calclcm(a, b)
    gcd = calcgcd(a, b)
    (a ÷ gcd) * b
end


function main()
    a, b = readline() |> converttointlist
    result = calclcm(a, b)
    println(result)
end


main()
